### DISCUSSION TITLE
Download all project SQL code at one time

### DISCUSSION THREAD

**Discussion by mkiang** [redivis/meta#177] [Feature requests]
**What is the problem you'd like us to address**
When a project has many transformations across many tables, getting a copy of the SQL code can be quite tedious. 

**Describe a solution (if you have one in mind)**
I would like to download all the SQL code at one time. This can be a zip file containing each transformation's SQL in a separate file, or (preferably) a large SQL script. I imagine even having the entire tree with metadata like table/transform names and relevant script for each could be encoded in a json file. 

**Describe other related products we could check out (if applicable)**

**Any additional context**

**Comment by imathews** [comment]
Thanks @mkiang. I like the idea of something along the lines of an .ipynb file which captures a snapshot of the entire project environment at that time. We'll keep you posted

**Comment by imathews** [comment]
@mkiang This functionality is now implemented in the [latest release](https://github.com/redivis/meta/releases/tag/0.20.0). From the main project screen, if you click the three-dot "More" button at top right, you can select "Download all code".

This will provide a JSON representation of your project at that point in time, including all transforms and their generated SQL, as well as their relationship to each other.

You can still download the complete SQL used to create any particular table in the project by selecting the table -> "More" -> "Download upstream SQL code"

**Reply by mkiang** [reply]
Amazing! And just in time as I'm about to dive into two more Optum projects... Thanks for adding this!
